src/main/scala/com.gu.gibbons/utils/UrlGenerator.scala

```diff
@@ -16,12 +17,17 @@ abstract class HashGenerator(md: MessageDigest) {
 }
 
 class UrlGenerator(settings: Settings, md: MessageDigest) extends HashGenerator(md) {
-  private def url(action: String, key: Key) =
-    s"${settings.bonoboUrl}/user/${key.consumerId}/${action}?${params(key, settings.salt)}"
+  private def keyUrl(action: String, key: Key) =
+    s"${settings.bonoboUrl}/user/${key.consumerId}/${action}?${keyParams(key, settings.salt)}"
+
+  private def userUrl(action: String, user: User) =
+    s"${settings.bonoboUrl}/user/${UserId.unwrap(user.id)}/${action}"
+
+  def extendKey(key: Key): String = keyUrl("extend", key)
 
-  def extendKey(key: Key): String = url("extend", key)
+  def deleteKey(key: Key): String = keyUrl("delete", key)
 
-  def deleteKey(key: Key): String = url("delete", key)
+  def deleteUserAndKey(user: User): String = userUrl("unverified", user)
```

Comment:
Perhaps this ought to be called `deleteUnverifiedUserAndKeys` if that's exactly what its use case is?